### PR TITLE
fix: translate type products as linear arrays

### PR DESCRIPTION
**Pull request by l29ah**
fixes https://github.com/TokTok/hs-msgpack-binary/issues/75

---
This change is [Reviewable](https://reviewable.io/reviews/toktok/hs-msgpack-types/34)

### CODE REVIEW

**Comment by iphydf** [comment]
The travis logs don't load for me right now. Are the tests passing?

**Comment by l29ah** [comment]
Yup, `cabal test` are.

**Comment by iphydf** [comment]
Ok, restarting the travis run. It says "null" when I try to load the log.

**Comment by iphydf** [comment]
Can you fix the style issue in the meantime? I think it's import alignment.

**Comment by l29ah** [comment]
On Mon, Jun 29, 2020 at 10:47:53AM -0700, iphydf wrote:
> Can you fix the style issue in the meantime? I think it's import alignment.

No. I find such alignment rules worthless and following them a dumb time waste.

-- 
()  ascii ribbon campaign - against html mail
/\  http://arc.pasp.de/   - against proprietary attachments

File: src/Data/MessagePack/Types/Generic.hs [hunk 1]
```diff
@@ -28,9 +30,11 @@ instance GMessagePack U1 where
     gFromObject ObjectNil = return U1
     gFromObject _         = fail "invalid encoding for custom unit type"
 
-instance (GMessagePack a, GProdPack b) => GMessagePack (a :*: b) where
+instance (GProdPack a) => GMessagePack (a) where
```
Comment (iphydf) [inline src/Data/MessagePack/Types/Generic.hs:33]:
```
./src/Data/MessagePack/Types/Generic.hs:33:40: Warning: Redundant bracket
Found:
  (a)
Perhaps:
  a
```

**Review by iphydf — APPROVED** [review]

**Pull request merged**